ceres.py

```diff
@@ -473,27 +499,60 @@ def read(self, fromTime, untilTime):
         except NoData:
           continue
 
+        # Calculate the cost for growing values, and aggregate current data if too big
+        maxCost = retentions.get(timeStep, 0)
+        missingGap = (requestUntilTime - series.endTime)
+        while len(bigger_timeSteps) and maxCost != 0:
+          newValues = int(len(series.values) * (series.timeStep / timeStep))
```

Comment:
Thinking aloud: are we relying on carbon's storage-schema validation to enforce retentions being evenly divisible? 
TBC, I dont think we should do any validation here but if it's not done somewhere I expect we'll eventually see some weird bug reports when people configure retentions unusually